src/components/UpdateCredit.jsx

```diff
@@ -1,21 +1,36 @@
 import React from "react";
 import { useState, useEffect } from "react";
 import { useStateContext } from "../contexts/ContextProvider";
+import { useSelector, useDispatch } from "react-redux";
 import Button from "./Button";
 import { Stripe_KEY } from "../Address";
 import { loadStripe } from "@stripe/stripe-js";
 import { CardElement, useStripe, useElements } from "@stripe/react-stripe-js";
 import { Elements } from "@stripe/react-stripe-js";
-import { addPaymentAuth } from "../api";
+import { addPaymentAuth, getUserInfo} from "../api";
+import { addPaymentMethod } from "../reducers/authSlicer";
 
 const stripePromise = loadStripe(Stripe_KEY);
 
 const UpdateCredit = () => {
   const { currentColor } = useStateContext();
-  const [cardNumber, setCardNumber] = useState("111 ");
+  const [cardNumber, setCardNumber] = useState("000");
+  const dispatch = useDispatch(); 
   const stripe = useStripe();
   const elements = useElements();
   const JWTtoken = localStorage.getItem("jwtToken");
+
+  useEffect(() => {
```

Comment:
There is another unused useEffect hook towards the end of the component, consider removing it if it's not needed.

Reply:
Removed, thanks for pointing out